bot/cogs/moderation/infractions.py

```diff
@@ -235,8 +235,32 @@ async def apply_ban(self, ctx: Context, user: UserSnowflake, reason: str, **kwar
 
         Will also remove the banned user from the Big Brother watch list if applicable.
         """
-        if await utils.has_active_infraction(ctx, user, "ban"):
-            return
+        # In the case of a permanent ban, we don't need get_active_infractions to tell us if one is active
+        send_msg = "expires_at" in kwargs
+        active_infraction = await utils.get_active_infractions(ctx, user, "ban", send_msg)
+
+        if active_infraction:
+            log.trace("Active infractions found.")
+            if (
+                active_infraction.get('expires_at') is not None
+                and kwargs.get('expires_at') is None
+            ):
+                log.trace("Active ban is a temporary and being called by a perma.  Removing temporary.")
+                await self.pardon_infraction(ctx, "ban", user, send_msg)
```

Comment:
I thought it's possible to edit the expiration via the API. Are you not doing that for the sake of keeping separate records?

Reply:
Yeah.  The intention is to make them separate infractions so that we can see what triggered the initial ban and what ended up breaking the camel's back causing the perma.